### ISSUE TITLE
[Bug] RandomBrightnessContrast mathematical formulation does not match the implementation

### ISSUE THREAD

**Issue by ternaus** [albumentations-team/AlbumentationsX#372]
Hi,

[`RandomBrightnessContrast`](https://albumentations.ai/explore/transform/RandomBrightnessContrast/) 2.3.7 documents torchvision-style brightness and contrast, but its implementation executes a different zero-centered affine transform. The fix should make the default model torchvision-compatible and preserve the current max-scaled affine model as an explicit option.

## Behavior in 2.3.7

The [2.3.7 docstring](https://github.com/albumentations-team/AlbumentationsX/blob/06e8d46a14292701b1b49f0ee20c8fde36f59207/albumentations/augmentations/pixel/color_basic.py#L717-L724) says contrast is centered around the image mean and brightness is multiplicative:

```text
1. Contrast adjustment: x' = clip((x - mean) * (1 + a) + mean)
2. Brightness adjustment:
   If brightness_by_max is True:  x'' = clip(x' * (1 + beta))
   If brightness_by_max is False: x'' = clip(x' + beta * max_value)
```

The [implementation](https://github.com/albumentations-team/AlbumentationsX/blob/06e8d46a14292701b1b49f0ee20c8fde36f59207/albumentations/augmentations/pixel/color_basic.py#L778-L796) instead runs one affine operation:

```python
max_value = MAX_VALUES_BY_DTYPE[img.dtype]
beta = beta * max_value if self.brightness_by_max else beta * float(mean(img))
return albucore.multiply_add(img, alpha, beta, inplace=False)
```

With `delta_b` sampled from `brightness_range`, `delta_c` sampled from `contrast_range`, and `c = 1 + delta_c`, the actual pre-clipping formulas are:

```text
brightness_by_max=True:  output = c * image + delta_b * dtype_max
brightness_by_max=False: output = c * image + delta_b * mean(image)
```

This reproducer shows the brightness branch reversal:

```python
import albumentations as A
import numpy as np

image = np.full((4, 4, 3), 100, dtype=np.uint8)

for brightness_by_max in (True, False):
    transform = A.RandomBrightnessContrast(
        brightness_range=(0.2, 0.2),
        contrast_range=(0.0, 0.0),
        brightness_by_max=brightness_by_max,
        p=1,
    )
    print(brightness_by_max, np.unique(transform(image=image)["image"]))
```

`albumentationsx==2.3.7` prints:

```text
True [151]
False [120]
```

## Resolution

Make `brightness_by_max=False` the default and give that mode torchvision semantics. Let `b = 1 + delta_b`, `c = 1 + delta_c`, and `mu` be the grayscale-luminance mean for RGB or the global mean for other channel counts:

```text
contrast = clip(c * image + (1 - c) * mu)
output = clip(b * contrast)
```

Keep `brightness_by_max=True` as the existing OpenCV-style affine model:

```text
output = clip(c * image + delta_b * dtype_max)
```

The implementation can reuse the same fused helper as `ColorJitter` for the default path:

```diff
- beta = beta * max_value if self.brightness_by_max else beta * float(mean(img))
- return albucore.multiply_add(img, alpha, beta, inplace=False)
+ if self.brightness_by_max:
+     return albucore.multiply_add(img, alpha, beta * max_value, inplace=False)
+
+ return fpixel.apply_brightness_contrast_torchvision(
+     img,
+     brightness_factor=1.0 + beta,
+     contrast_factor=alpha,
+     brightness_first=False,
+ )
```

The torchvision model is the better general-training default because its sampled controls have separate effects: brightness scales the mean intensity, while contrast changes deviations around the mean. The OpenCV-style model remains useful when the augmentation should simulate an additive black-level or sensor offset. Neither policy guarantees better validation accuracy for every dataset; the selected ranges still need to match plausible deployment variation.

## Compatibility and tests

This is intentionally a breaking default change. Code that needs the 2.3.7 default output can preserve it by setting `brightness_by_max=True`. The previous `brightness_by_max=False` behavior changes from a mean-scaled additive offset to torchvision-compatible multiplicative brightness and mean-centered contrast.

The regression coverage should include `uint8` and `float32`, RGB and arbitrary channel counts, per-image batch means, per-slice volume means, `ensure_safe_output`, invalid negative torchvision factors, direct output comparison with torchvision, and the preserved `brightness_by_max=True` formula. The ASV parameter matrix should retain separate torchvision and OpenCV cases because the mean-centered model performs an additional image reduction.
